lib/reek/ast/sexp_extensions/constant_assignment.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+module Reek
+  module AST
+    module SexpExtensions
+      # Utility methods for :casgn nodes.
+      module CasgnNode
+        def send_expression
+          children[2]
+        end
+
+        def ancestor
+          send_expression.value_assigned
+        end
+
+        def class_creation?
```

Comment:
You can use `defines_module?` from `lib/reek/ast/sexp_extensions/module.rb` to simplify this method.